app/scripts/models/auth_brokers/oauth-redirect.js

```diff
@@ -68,7 +69,19 @@ define(function (require, exports, module) {
           if (result.action) {
             extraParams['action'] = result.action;
           }
-          win.location.href = Url.updateSearchString(result.redirect, extraParams);
+
+          const redirect = Url.updateSearchString(result.redirect, extraParams);
+          const uap = new UserAgent(win.navigator.userAgent);
+
+          const isChromeAndroid = uap.isChromeAndroid();
+          if (isChromeAndroid) {
+            // For OAuth Applications redirection using Chrome Android,
+            // the action must be triggered by the user, not Javascript.
+            // See #6089
+            return redirect;
```

Comment:
somehow this needs to detect a `afterSignInConfirmationPoll` or `afterSignUpConfirmationPoll` or other polls

Reply:
if it's not a poll we can just do the old redirect and not show the button

Reply:
I wonder if it would be useful to factor this out into a "oauth-redirect-android.js" auth_broker, which might give you a bit of leeway with detecting `afterSignInConfirmationPoll` etc.